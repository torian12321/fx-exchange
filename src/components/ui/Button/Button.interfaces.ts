export interface IButton {
  children?: any;
  caption?: string;
  alt?: string;
  className?: string;
  disabled?: boolean;
  loading?: boolean;
  onClick?: any;
  type?: "reset" | "submit" | "button";
  form?: string;
}
