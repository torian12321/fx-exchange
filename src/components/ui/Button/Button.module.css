.wrapper {
  font-size: 12px;
  padding: .4em 2em;
  border-radius: 4px;
  border: none;
  box-sizing: border-box;

  background: #b159a4;
  color: white;
}

.loader {
  font-size: 1em;
  color: white;
  display: block;
  margin: auto;

  position: relative;
  top: 0%;
}

.wrapper_IsLoading > .content {
  opacity: 0;
  height: 0;
  display: block;
}