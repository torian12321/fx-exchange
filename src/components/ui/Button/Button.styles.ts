import { css } from '@emotion/core'
import { theme, functions } from 'assets/styles';

const color = theme.colors.secondary;

export const btn = css`
  outline: none;
  font-size: 1rem;
  padding: .4em 2em;
  border-radius: 10em;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  line-height: 2em;

  background: ${color};
  color: ${functions.smartTextColor(color)};

  &:hover {
    background: ${functions.darkenColor(color, 5)};
  }
  &:focus,
  &:active {
    background: ${functions.darkenColor(color, 10)};
  }
`

export const btnDisabled = css`
  &,
  &:hover {
    cursor: not-allowed;
    background: ${theme.colors.states.disabled.bg};
    color: ${theme.colors.states.disabled.text};
  }
`

export const loader = css`
  font-size: 1em;
  color: white;
  display: block;
  margin: auto;

  position: relative;
  top: 0%;
`;


export const contentLoading = css`
  opacity: 0;
  height: 0;
  display: block;
`;
