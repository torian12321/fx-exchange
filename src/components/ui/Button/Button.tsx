import React from 'react';
// import logo from './logo.svg';
// import './App.css';

const Button: React.FC = () => {
  return (
    <button>
      bla bla
    </button>
  );
}

export { Button };
