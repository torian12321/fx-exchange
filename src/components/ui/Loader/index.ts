import { sbIndex as index } from 'components/ui';
import { Spinner } from './Spinner';

const sbIndex = `${index}Loaders/`;

export default Spinner;
export { sbIndex };
export { Spinner }
export { Ellipsis } from './Ellipsis';
