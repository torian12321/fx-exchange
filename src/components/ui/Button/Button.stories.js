import React from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import { Button } from './Button';

export default {
  title: 'Components|Atoms/Button',
  decorators: [withKnobs],
};

export const basic = () => <Button onClick={action('clicked')} caption='Hello Button' />;

export const customContent = () => (
  <Button
    onClick={action('clicked')}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
);

export const outline = () => (
  <>
    <Button onClick={action('clicked')} caption='Outline => false' />
    <Button onClick={action('clicked')} caption='Outlined' outline />
  </>
);

export const disabled = () => (
  <>
    <Button onClick={action('clicked')} caption='Hello Button' />
    <Button onClick={action('clicked')} caption='Hello Button' /><Button onClick={action('clicked')} caption='Hello Button' /><Button onClick={action('clicked')} caption='Hello Button' /><Button onClick={action('clicked')} caption='Hello Button' />
    <Button onClick={action('clicked')} caption='Disabled button' disabled />
  </>
);
export const loading = () => (
  <>
    <Button onClick={action('clicked')} caption='Hello Button' />
    <Button onClick={action('clicked')} caption='Hello Button' isLoading />
  </>
);

export const withVars = () => (
  <Button
    outline={boolean("Outline", false)}
    disabled={boolean("Disabled", false)}
    isLoading={boolean("isLoading", false)}
    caption={text("Caption", "Cutom button")}
    onClick={action('clicked')}
  />
);
