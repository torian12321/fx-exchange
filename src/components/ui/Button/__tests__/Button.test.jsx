import React from "react";
import { create } from "react-test-renderer";
import * as sinon from "sinon";
import { Ellipsis } from "../../Loader";
import { Button } from '../Button';

describe("UI Component: Button", () => {
  it(`should render as a button`, () => {
    const component = create(
      <Button caption="Button text" />
    );

    expect(component.toJSON().type).toEqual('button');
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should be clickable ", () => {
    // Given
    const fn = sinon.stub();
    const component = create(
      <Button caption="Button text" onClick={fn} />
    );

    // When
    component.root.findByType('button').props.onClick();

    // Then
    expect(component.toJSON()).toMatchSnapshot();
    expect(fn.callCount).toBe(1);
  });

  describe("disalbed", () => {
    it("should be disabled", () => {
      // Given
      const component = create(
        <Button caption="Button text" disabled />
      );

      // When
      const btn = component.root.findByType('button');

      // Then
      expect(component.toJSON()).toMatchSnapshot();
      expect(btn.props.disabled).toBe(true);
    });
    it("should be non clickable", () => {
      // Given
      const fn = sinon.stub();
      const component = create(
        <Button caption="Button text" onClick={fn} disabled />
      );

      // When
      const btn = component.root.findByType('button');
      btn.props.onClick();

      // Then
      expect(fn.callCount).toBe(0);
    });
  });


  describe("isLoading", () => {
    it(`should render with Ellipses`, () => {
      // Given && When
      const component = create(
        <Button caption="Button text" isLoading />
      );

      const btn = component.root;

      // Then
      expect(component.toJSON()).toMatchSnapshot();
      expect(btn.findAllByType("button").length).toEqual(1);
      expect(btn.findAllByType(Ellipsis).length).toEqual(1);
    });
    it(`should not be clickable`, () => {
      // Given
      const fn = sinon.stub();
      const component = create(
        <Button caption="Button text" isLoading onClick={fn} />
      );

      // When
      const btn = component.root.findByType('button');
      btn.props.onClick();

      // Then
      expect(fn.callCount).toBe(0);
      expect(btn.props.disabled).toBe(true);
    });
  });
});
