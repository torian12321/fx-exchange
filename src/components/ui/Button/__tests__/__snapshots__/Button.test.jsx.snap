// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component: Button disalbed should be disabled 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1dis8g1-Button"
  >
    Button text
  </span>
</button>
`;

exports[`UI Component: Button disalbed should be non clickable 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1dis8g1-Button"
  >
    Button text
  </span>
</button>
`;

exports[`UI Component: Button isLoading should not be clickable 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1lzay8-Button"
  >
    Button text
  </span>
  <div
    className="css-sjaooz-Button"
  >
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
  </div>
</button>
`;

exports[`UI Component: Button isLoading should render Ellipses 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1lzay8-Button"
  >
    Button text
  </span>
  <div
    className="css-sjaooz-Button"
  >
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
  </div>
</button>
`;

exports[`UI Component: Button isLoading should render with Ellipses 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1lzay8-Button"
  >
    Button text
  </span>
  <div
    className="css-sjaooz-Button"
  >
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
  </div>
</button>
`;

exports[`UI Component: Button should be clickable  1`] = `
<button
  aria-label="Button text"
  className=" css-mslgpb-Button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1dis8g1-Button"
  >
    Button text
  </span>
</button>
`;

exports[`UI Component: Button should be non clickable when disabled 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1dis8g1-Button"
  >
    Button text
  </span>
</button>
`;

exports[`UI Component: Button should render 1`] = `
<button
  aria-label="Button text"
  className=" css-mslgpb-Button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1dis8g1-Button"
  >
    Button text
  </span>
</button>
`;

exports[`UI Component: Button should render Ellipses when isLoading 1`] = `
<button
  aria-label="Button text"
  className=" css-w1zx51-Button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1lzay8-Button"
  >
    Button text
  </span>
  <div
    className="css-sjaooz-Button"
  >
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
    <span
      className="css-1rppay8-Ellipsis"
    />
  </div>
</button>
`;

exports[`UI Component: Button should render as a button 1`] = `
<button
  aria-label="Button text"
  className=" css-mslgpb-Button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    className="css-1dis8g1-Button"
  >
    Button text
  </span>
</button>
`;
